### PR TITLE
Stun Gun adjustments

### PR DESCRIPTION
**Pull request by anoobindisguise**
#### Summary
Balance "Rebalance stun gun power consumption and effectiveness"

#### Purpose of change
Adjustments according to my research in #53127.

#### Describe the solution
Stun guns and weapons based on them use 2kJ per zap. I chose this number because Cata does not handle decimal kJ in batteries and 250 uses per medium battery remains the same as the "hundreds of uses" described in my research; while the consumption is much higher the ratio is the same. I included extra flavor text to justify these models being so much stronger than IRL versions.
Also reduced player stun gun damage from 5, 25 to 3, 16.
Finally, the player stun gun attack could be dodged, but the enemy stun gun attack could not - I adjusted this so that you have a 50% chance to avoid the attack at 10 dodge. As it is a melee attack being driven into you by a robot or a feral, it is reasonable that it can be avoided.

#### Describe alternatives you've considered
Currently the monster stun gun attack is hardcoded as 10d10 attack roll, it could be made variable. The difference between specific enemies is minimal however.
Damage dealt by player stun gun might still be too high. Currently you can expect a kill in about 5 seconds of continuous stunning on a usual zombie (compared to 3 seconds or so before) which feels OK for what is basically a handheld electric chair deployed against "rioters", but it's tough to tell. Leaving the damage as normal could also be ok as 2kJ is an extremely high amount of charge regardless.

### CODE REVIEW

**Comment by kevingranade** [comment]
If you're rationalizing the stungun it needs to:
Do no damage.
Require a 5 second application at the power drain you found to inflict a longer term debilitating effect.
Be stopped by sufficient clothing thickness.
Scale based on target size, creatures meaningfully larger  than humans are not going to be affected the sane.

Basically it either needs to be a long duration action that stunlocks the target while it's being applied, or an immediate action that prevents the target from taking its next turn (I e. Set its moves to -speed) and inflicts an effect that accumulates in intensity and inflicts a longer term stun once it reaches higher intensity.

**Comment by anoobindisguise** [comment]
OK, currently the player stun gun works like this:
- check target dodge and target bash armor vs player's melee skill.
- deduct moves based on the size of the creature. tiny creatures lose the most, huge creatures don't lose enough to cover one turn.
- if target moves are sufficiently in the red start doing damage to them.

Will try to apply this behavior to skitterbots. The ramping stun style doesn't really work against a player character.

**Comment by kevingranade** [comment]
No damage means no damage, a stun gun is not reliably lethal, period.

**Comment by anoobindisguise** [comment]
Ok, couple questions - 
Stun gun can't kill due to low amperage, but a 1-2kJ model would have much higher amperage and could be lethal. I can remove damage if you want it's just a consideration due to the circumstances of the item in game (also can't make it use just 25j due to battery limitations)
what do you want to do about skitterbots and other stunner enemies? Make them solely a debuffer enemy that causes pain and slowdown?

File: data/json/items/melee/misc.json [hunk 1]
```diff
@@ -60,7 +60,7 @@
     "type": "TOOL",
     "category": "weapons",
     "name": { "str": "stun gun" },
-    "description": "This is a high-powered stun gun.  Use this item to attempt to electrocute an adjacent enemy, damaging and temporarily paralyzing them.  Because the shock can actually jump through the air, it is difficult to miss.",
+    "description": "This is a high-powered stun gun.  Carrying nearly a hundred times the shocking power of traditional models, this was made to \"nonlethally subdue\" rioters in the early days of the Cataclysm and it induces paralysis, extreme pain and even cardiac arrest in short order.  Use this item to attempt to electrocute an adjacent enemy, damaging and temporarily paralyzing them.  Because the shock can actually jump through the air, it is difficult to miss.",
```
Comment (kevingranade) [inline data/json/items/melee/misc.json:63]: Nope, regular stun gun.
```suggestion
    "description": "This is a high-powered stun gun.  Use this item to attempt to electrocute an adjacent enemy, damaging and temporarily paralyzing them.",
```

File: src/iuse.cpp [hunk 2]
```diff
@@ -3922,25 +3922,40 @@ cata::optional<int> iuse::tazer( Character *p, item *it, bool, const tripoint &p
...
+                                  target->disp_name() );
+        if( target->moves < -250 ) {
+            target->deal_damage( p, bodypart_id( "torso" ), damage_instance( damage_type::ELECTRIC,
+                                 rng( 2,
+                                      10 ) ) ).total_damage();
+            p->add_msg_player_or_npc( m_good,
+                                      _( "You harshly shock %s!" ),
+                                      _( "<npcname> harshly shocks %s!" ),
                                       target->disp_name() );
         }
```
Comment (kevingranade) [inline src/iuse.cpp:3960]: No damage.
```suggestion
        p->add_msg_player_or_npc( m_good,
                                  _( "You shock %s!" ),
                                  _( "<npcname> shocks %s!" ),
                                  target->disp_name() );
```

**Comment by anoobindisguise** [comment]
Ok, so what would you recommend I do about skitterbot damage, power usage? Leave it at 1kJ like all other "abstracted" power costs? (Such as talking dolls costing 1kJ)

**Comment by kevingranade** [comment]
The stun gun is not nonlethal because you can't make a lethal electric weapon, it's nonlethal because it's designed to be nonlethal.
Skitterbots are a whole different thing, they may or may not be intended to be lethal, they may or may not need to even exist.

**Comment by anoobindisguise** [comment]
I absolutely agree a normal stun gun is not lethal, it makes sense a skitterbot may be designed to be lethal where a normal one is not, I'll change that back to 1-5 damage. Question is, do we want to have "abnormal" stun guns, that can kill? If so, should those be an off-market, non-looted variety? Or would police deploy them against rioters?

**Comment by kevingranade** [comment]
I'm not aware of anyone making lethal stun guns.  There are already person portable lethal devices of all kinds that don't need batteries.

**Pull request merged**